desk/ted/vitals/connection-check.hoon

```diff
@@ -45,23 +46,30 @@
   ::  if we're a moon, check if we can contact our planet
   ::
   ::  NN: failing to contact our sponsor is only a failure condition for moons,
-  ::      since currently only moons receive additional routing help from their
-  ::      sponsors
+  ::      since currently only moons require the direct sponsor to be online for
+  ::      peers to grab the moon keys
   ::  NN: we do this after the initial target check because if we're a moon and
   ::      our planet is down, it's useful to talk to ships that still have live
   ::      wires (e.g. for troubleshooting); thus, by waiting to perform this
   ::      check, we don't report %no-our-planet for every connectivity check
   ::      when attempting to track down a live peer from whom to seek help
   ::
-  ;<    sqos=qos:ames
+  ;<  moon-sponsor-reachable=?
       bind:m
-    =/  mm  (strand ,qos:ames)
+    =/  mm  (strand ,?)
     ^-  form:mm
     ?.  ?=(%earl (clan:title our))
-      (pure:mm [%live *@da])
-    (scry:io qos:ames ~[%gx %vitals %sponsor %vitals-qos])
-  ?.  ?=(%live -.sqos)
-    (post-result [%no-our-planet last-contact.sqos])
+      (pure:mm %.y)
+    =/  sponsor=@p  (end 5 our)
+    ;<  ~  bind:mm  (update-status [%trying-sponsor sponsor])
+    ;<  pchek=(unit)   bind:mm  (check-online sponsor target-timeout:vitals)
+    ?:  ?=([%$ %$] pchek)
```

Comment:
The `[%$ %$]` feels weird here. At the very least should be `[~ %$]`, because we know it's `(unit)`. Maybe even `[~ *]` or just `^` (which then becomes a `?^`), given we don't care about the contents.

I _think_ `+check-online` also only ever returns `~` or `[~ ~]`, but for some reason its type doesn't explicate that. So strange.